src/state.js

```diff
@@ -22,11 +22,11 @@ var state = {
 
   /**
    * @param {boolean} strict - When `true`, only consider ES6 when in
-   *                           "esversion: 6" code and *not* in "moz".
+   *                           "esversion: 6" code.
    */
   inES6: function(strict) {
     if (strict) {
-      return !this.option.moz && this.option.esversion === 6;
+      return this.option.esversion === 6;
     }
     return this.option.moz || this.option.esversion >= 6;
```

Comment:
I agree, but disabling it now would break backwards compatibility. The nice thing is that [we've been careful to introduce more recent ES6 features like `new.target` using the "strict" setting](https://github.com/jshint/jshint/blob/46db9238da535a2d82d7565deac887de6e602ad4/src/jshint.js#L2342-L2344), which encourages new codebases to configure *both* `moz: true` and `esnext: true`